### PR TITLE
Added proxy api for Plausible on current express server

### PR DESCRIPTION
**Pull request by yathomasi**
Fixes https://github.com/iterative/iterative.ai/issues/423

Created proxy API for plausible basically to handle these two requests
```
pl/js/script.js -> https://plausible.io/js/script.js
pl/event    -> https://plausible.io/api/event
```

Todo: 
- [x] get this API implementation reviewed and finalize
- [x] replace current plausible API with this proxy API

**Update**: 
Script: https://dvc-org-proxy-plausible-cd4crn.herokuapp.com/api/pl/js/script.js
Event API: https://dvc-org-proxy-plausible-cd4crn.herokuapp.com/api/pl/event

### CODE REVIEW

**Review by rogermparent — COMMENTED** [review]
The core concept looks solid to me! Most of our APIs have a caching system using `node-cache`, could that be valuable here as well?

File: src/server/middleware/api/plausible.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+const { isProduction } = require('../../utils')
+const NodeCache = require('node-cache')
+require('isomorphic-fetch')
+
+const ScriptName = '/pl/script.js'
```
Comment (julieg18) [inline src/server/middleware/api/plausible.js:5]: Why are these variables capitalized? If they're constants, shouldn't they be completely uppercase 🤔

**Review by julieg18 — COMMENTED** [review]
Looks good to me, but I have little experience in NodeJS. @rogermparent, what do you think?

File: src/server/middleware/api/plausible.js (hunk 1 above)
Comment (rogermparent) [reply src/server/middleware/api/plausible.js:5]: That said, maybe these constants should be shared with the other files that require them? [DefaultSEO](https://github.com/iterative/dvc.org/pull/3423/files#diff-aa1a28d5dbb813758bec2678c00fcb7cf9104a2b5cfa9bbe8313270feaeec84a) could use this link.

**Review by rogermparent — COMMENTED** [review]
I'm admittedly not the strongest in Express, but nothing looks outright obviously here. I think a change to the consts is all we need to make this mergeable!

File: src/server/middleware/api/plausible.js (hunk 1 above)
Comment (yathomasi) [reply src/server/middleware/api/plausible.js]: Thanks for pointing it out.

**Comment by yathomasi** [comment]
It looks like the API is not working on gatsby cloud deployment. If we are going to use Gatsby Cloud, it would be better to use [Gatsby Functions](https://www.gatsbyjs.com/docs/reference/functions/)

**Comment by rogermparent** [comment]
We've mostly just enabled Gatsby Cloud to give content editors deploy previews while Heroku integration is busted. We certainly could use functions, but I figured we'd only go down that road if we wanted a full switch over.

**Review by rogermparent — APPROVED** [review]
Even if the deploy previews can't test it, I think it's fine to bite the bullet and try this in prod as long as we ensure it works soon after it's merged.

**Pull request merged**